scripts/validate_docstrings.py

```diff
@@ -490,10 +491,41 @@ def yields(self):
     @property
     def method_source(self):
         try:
-            return inspect.getsource(self.obj)
+            source = inspect.getsource(self.obj)
+            # Remove common indentation.
+            first_spaces = re.match(' +', source)
+            if first_spaces:
+                first_spaces = first_spaces.group(0)
+                source = re.sub('^' + first_spaces, '', source)
+                source = re.sub(r'\n' + first_spaces, r'\n', source)
+            return source
         except TypeError:
             return ''
 
+    @property
+    def method_returns(self):
+        tree = ast.parse(self.method_source).body
+        if tree:
+            root = tree[0]
+
+            def gather_returns(node):
+                gathered = [node] if isinstance(node, ast.Return) else []
+                for child in ast.iter_child_nodes(node):
+                    # Ignore nested functions and its subtrees.
+                    if not isinstance(child, ast.FunctionDef):
+                        gathered.extend(gather_returns(child))
+                return gathered
+            # Walk the tree recursively and gather the return nodes.
+            return_values = [r.value for r in gather_returns(root)]
```

Comment:
As opposing to using `ast.walk()` to iterate over all the nodes on the method's AST, here I'm recursively walking the tree myself.

The reason for that is to respect cases like the following:
```
def A():
    def B():
        return "Hello World!"
    print(B())
    return None
```
In this case, there is a inner function which returns a string, but the function being evaluated itself always return None, thus shouldn't have a Returns section on the docstring.

If I read all the nodes indiscriminately, as happens with `ast.walk()`, I can't distinguish returns inside nested functions from returns on the function actually being evaluated. If I search trough it myself, though, I can prune the recursion when I reach a (nested) function node, thus ignoring it's entire subtree and possible return nodes in it.